src/android/TP6mAggregate.jsx

```diff
@@ -15,27 +17,108 @@ import { getData } from '../vendor/perfherder';
 import { withErrorBoundary } from '../vendor/errors';
 import jx from '../vendor/jx/expressions';
 import { Cube, HyperCube, window } from '../vendor/jx/cubes';
-import { g5Reference, TARGET_NAME } from './config';
+import { TARGET_NAME } from './config';
 import ChartJSWrapper from '../vendor/components/chartJs/ChartJsWrapper';
 import timer from '../vendor/timer';
 import { DetailsIcon } from '../utils/icons';
+import { TimeDomain } from '../vendor/jx/domains';
+
+
+const DESIRED_TESTS = ['cold-loadtime'];
+const DESIRED_PLATFORMS = ['p2-aarch64', 'g5'];
+const DESIRED_BROWSER = ['fenix'];
+const REFERENCE_BROWSER = ['fennec64'];
 
 /*
 condition - json expression to pull perfherder data
  */
 async function pullAggregate({
   condition,
-  tests,
-  sites,
-  platforms,
+  test,
+  platform,
   timeDomain,
 }) {
+  const tests = selectFrom(TP6_TESTS).where({ test });
+  const testMode = tests.select('mode').first();
+  const sites = TP6M_SITES.filter(({ mode }) => mode.includes(testMode)).materialize();
+  const platforms = selectFrom(BROWSER_PLATFORMS).where({ platform, browser: DESIRED_BROWSER });
   const readData = timer('read data');
-  const sources = await getData(condition);
+  const referenceRange = new TimeDomain({ past: 'month', ending: 'today' });
+  const referencePlatforms = selectFrom(BROWSER_PLATFORMS).where({ platform, browser: REFERENCE_BROWSER });
+  const referenceFilter = TP6_COMBOS
+    .where({
+      browser: REFERENCE_BROWSER,
+      platform,
+      test,
+    })
+    .select('filter')
+    .toArray();
+  const [fennec64, sources] = await Promise.all([
+    getData({ or: referenceFilter }),
+    getData(condition),
+  ]);
 
   readData.done();
 
   const processData = timer('process data');
+
+  const reference = selectFrom(fennec64)
```

Comment:
@arnold-iakab @airimovici    Notice this block is near-identical to `measured = ...` directly below: Both the reference data and the measured data is processed the same.  This duplication is a bit unfortunate; if it happens a third time, then it belongs in a function.